test/views/DeductionsFromPaymentsViewSpec.scala

```diff
@@ -0,0 +1,171 @@
+  }
+
+  object ExpectedAgentCY extends SpecificExpectedResults {
+    override val expectedTitleText: String = "A yw’ch cleient wedi cael didyniadau CIS o’i daliadau gan gontractwyr?"
+    override val expectedErrorTitleText: String = s"Error: $expectedTitleText"
+    override val expectedParagraphText: String = "Cewch hyd i’r didyniadau mewn datganiadau a roddwyd i’ch cleient gan gontractwyr."
+    override val expectedErrorText: String = "Dewiswch ‘Iawn’ os oedd gan eich cleient ddidyniadau CIS wedi'u cymryd o'i daliadau gan gontractwyr."
+  }
+
+  override protected val userScenarios: Seq[UserScenario[CommonExpectedResults, SpecificExpectedResults]] = Seq(
+    UserScenario(isWelsh = false, isAgent = false, CommonExpectedEN, Some(ExpectedIndividualEN)),
+    UserScenario(isWelsh = false, isAgent = true, CommonExpectedEN, Some(ExpectedAgentEN)),
+    UserScenario(isWelsh = true, isAgent = false, CommonExpectedCY, Some(ExpectedIndividualCY)),
+    UserScenario(isWelsh = true, isAgent = true, CommonExpectedCY, Some(ExpectedAgentCY))
+  )
+
+  private lazy val underTest = inject[DeductionsFromPaymentsView]
+
+  userScenarios.foreach { userScenario =>
+    s"language is ${welshTest(userScenario.isWelsh)} and request is from an ${agentTest(userScenario.isAgent)}" should {
+      "render page with empty form and no value selected" which {
+        implicit val userSessionDataRequest: AuthorisationRequest[AnyContent] = getAuthRequest(userScenario.isAgent)
+        implicit val messages: Messages = getMessages(userScenario.isWelsh)
+
+        val pageModel = aDeductionFromPaymentsPage.copy()
+        implicit val document: Document = Jsoup.parse(underTest(pageModel).body)
+
+        welshToggleCheck(userScenario.isWelsh)
+        titleCheck(userScenario.specificExpectedResults.get.expectedTitleText)
+        captionCheck(userScenario.commonExpectedResults.expectedCaption(taxYearEOY))
+        h1Check(userScenario.specificExpectedResults.get.expectedTitleText)
+        textOnPageCheck(userScenario.specificExpectedResults.get.expectedParagraphText, Selectors.paragraphTextSelector)
+        radioButtonCheck(userScenario.commonExpectedResults.expectedYesText, radioNumber = 1, checked = false)
+        radioButtonCheck(userScenario.commonExpectedResults.expectedNoText, radioNumber = 2, checked = false)
+        formPostLinkCheck(DeductionsFromPaymentsController.submit(taxYearEOY).url, Selectors.continueButtonFormSelector)
+        buttonCheck(userScenario.commonExpectedResults.expectedButtonText, Selectors.buttonSelector)
+      }
+
+      "render page with filled in form using selected 'Yes' value" which {
+        implicit val userSessionDataRequest: AuthorisationRequest[AnyContent] = getAuthRequest(userScenario.isAgent)
+        implicit val messages: Messages = getMessages(userScenario.isWelsh)
+
+        val form = new FormsProvider().deductionFromPaymentsForm(userScenario.isAgent)
```

Comment:
You need the form on every test. You can extract it on top of the test to avoid some duplication.

Reply:
The form calls are different:
https://github.com/hmrc/income-tax-cis-frontend/blob/cf3f66109f06a2e8ffc7dcfcf72cfaaec6d10773/test/views/DeductionsFromPaymentsViewSpec.scala#L134
https://github.com/hmrc/income-tax-cis-frontend/blob/cf3f66109f06a2e8ffc7dcfcf72cfaaec6d10773/test/views/DeductionsFromPaymentsViewSpec.scala#L153

But the code could be made shorter. 

A single declaration can be added just below the `userScenarios.foreach { userScenario =>` line:
`val form = new FormsProvider().deductionFromPaymentsForm(userScenario.isAgent)`

And then the separate `pageModel` declarations can be updated to the following:
`val pageModel = aDeductionFromPaymentsPage.copy(form = form.fill(value = true))`
`val pageModel = aDeductionFromPaymentsPage.copy(form = form.bind(Map(YesNoForm.yesNo -> "")))`